### ISSUE TITLE
Warn if conversion depends on the order in which interfaces are specified in class base

### ISSUE THREAD

**Issue by alrz** [dotnet/roslyn#58258]
```cs
using System;
I<string> a = new C();
Console.Write(a.M("")); // could print either "IConvertible" or "ICloneable"; should warn?

interface I<in T> where T : class
{
  string M(T arg);
}

class C : I<IConvertible>, I<ICloneable> // swap for a different output
{
  public string M(IConvertible arg) => nameof(IConvertible);
  public string M(ICloneable arg) => nameof(ICloneable);
}

**Comment by jaredpar** [comment]
I believe there are a couple of cases where you can end up with this type of ambiguous behavior. @AlekseyTs will likely remember the details here but these type of bugs have been around for ~15 years now. Guessing we considered a warning at the time and rejected it. Adding one now is likely to be a significant breaking change, not sure if it would rise to the level of a warning wave.

**Comment by AlekseyTs** [comment]
I believe this isn't a bug, but a known runtime behavior due to an ambiguity when variance is involved. VB reports a warning for type declarations like that:
```
warning BC42333: Interface 'I(Of System.ICloneable)' is ambiguous with another implemented interface 'I(Of System.IConvertible)' due to the 'In' and 'Out' parameters in 'Interface I(Of In T As Class)'.
```
It doesn't report anything for conversions.

**Comment by jaredpar** [comment]
Closing for now as I suspect that this is not going to meet the bar for a warning wave. Can re-consider if we see more evidence about it.
